src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Lower.java

```diff
@@ -1794,18 +1794,22 @@ JCStatement initField(int pos, Symbol rhs, Symbol lhs) {
                     make.Ident(rhs)).setType(lhs.erasure(types)));
     }
 
-    /** Return tree simulating the assignment {@code this.this$n = this$n}.
-     */
-    JCStatement initOuterThis(int pos, VarSymbol rhs) {
+    /** Return tree simulating null checking outer this and assigning. */
```

Comment:
probably this comment should state that the assignment is generated or not depending on the value of argument `stores`